core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -162,5 +198,8 @@ class HadoopMapReduceCommitProtocol(jobId: String, path: String)
       val tmp = new Path(src)
       tmp.getFileSystem(taskContext.getConfiguration).delete(tmp, false)
     }
+    if (runtimeOverwritePartition) {
```

Comment:
If we just read this function without the context of this PR, I might ask why we drop the staging directory only when `runtimeOverwritePartition ` is true? 

Any reason we want to keep it unchanged when `runtimeOverwritePartition ` is false?